enterprise_subsidy/apps/fulfillment/api.py

```diff
@@ -70,6 +70,9 @@ def get_smarter_client(self):
             api_url=settings.GET_SMARTER_API_URL
         )
 
+    def get_enterprise_client(self, transaction):
+        return transaction.ledger.subsidy.enterprise_client
```

Comment:
i am thinking that we may want to re-use the subsidy API client getters, so we can better control caching, etc?

Reply:
LGTM

[non-blocking aside] We do a lot of caching client getters inside model/viewset instance methods, but have we considered instead caching API keys globally within the client module so that they transparently persist across the whole request?  I feel like the current paradigm of cached client getters inside instances methods have a couple downsides:

* unit testing: can make it difficult to mock a client because there could be multiple different classes within which the cached getter is defined, and they all have to be tracked down and mocked or else the client will not appear consistent, and sometimes the cache could cause bleed between subsequent unit tests (especially if the client module is patched instead of the instance getter, an easy mistake to make).
* every new viewset or model class has to independently define a cached client getter, and potentially even duplicate logic to read settings to get an API version number.

I feel like there must be a good, robust way of designing a client that is self-caching across the whole request, and also is a single point of mocking for unit testing.  Thoughts?

Reply:
Yup, an idea I wanted to bring up for discussion tomorrow is "caching at the boundaries of the system", i.e. cache our views/outputs for the sake of clients; cache replicated data (our inputs) right when we read that data.  
I think we've brought up the pattern of a read-through cache before, which should align with this idea.
https://hazelcast.com/blog/a-hitchhikers-guide-to-caching-patterns/

Reply:
Another idea to bring up: persist cached data in models rather than in a cache-proper, so that we can introspect the state of cached data more easily, or even have something like a "schema" that describes the shape of the data.
We have one ok-ish pattern for this in https://github.com/openedx/enterprise-catalog/blob/master/enterprise_catalog/apps/api_client/enterprise_cache.py